src/aggregators/sortingdirect.jl

```diff
@@ -0,0 +1,142 @@
+# Sorting Direct Method, implementation following McCollum et al, 
+# "The sorting direct method for stochastic simulation of biochemical systems with varying reaction execution behavior"
+# Comp. Bio. and Chem., 30, pg. 39-49 (2006). 
+
+mutable struct SortingDirectJumpAggregation{T,S,F1,F2,RNG,DEPGR} <: AbstractSSAJumpAggregator
+    next_jump::Int
+    next_jump_time::T
+    end_time::T
+    cur_rates::Vector{T}
+    sum_rate::T
+    ma_jumps::S
+    rates::F1
+    affects!::F2
+    save_positions::Tuple{Bool,Bool}
+    rng::RNG
+    dep_gr::DEPGR
+    jump_search_order::Vector{Int}
+    jump_search_idx::Int
+    SortingDirectJumpAggregation{T,S,F1,F2,RNG}(nj::Int, njt::T, et::T, crs::Vector{T}, sr::T, maj::S,
```

Comment:
why as an inner constructor?

Reply:
The previous jump aggregations had needed this setup to avoid the outer constructor getting shadowed. I guess this may not be an issue here. I'll look into this and the `p` notation in ssajump.jl with my next PR (probably next reaction method).